app/services/master_controller.py

```
@@ -1524,49 +1526,129 @@ async def get_system_status(self, user_id: str) -> Dict[str, Any]:
+                            # Calculate metrics from snapshots
+                            history_points = [{
+                                "timestamp": snapshot.snapshot_date.strftime("%H:%M"),
+                                "portfolio_value_usd": float(snapshot.total_value_usd)
+                            } for snapshot in snapshots]
+                            
+                            # Get actual trades for the last 24 hours to count trades and PnL
+                            trades_24h_stmt = select(Trade).where(
+                                and_(
+                                    Trade.user_id == user_id,
+                                    Trade.executed_at >= start_time,
+                                    Trade.status == "completed"
```

Ruff (0.12.2)

Undefined name `Trade`

(F821)